### PR TITLE
update w_pass

### PR DESCRIPTION
**Pull request by harrison4ride**

### CODE REVIEW

File: projects/clib/build_w_pass.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+cd clib
+make -j$(nproc)
+
+# ! add Pass before building the fazzers
+export REPORT_FLAGS="-Xclang -load -Xclang ./ReportFunctionExecutedPass/libReportPass.so -flegacy-pass-manager"
```
Comment (EYH0602) [inline projects/clib/build_w_pass.sh]: these flags should be added before `make -j$(nproc)` for the PASS to work

File: projects/clib/build_w_pass.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+$CXX $CXXFLAGS $LIB_FUZZING_ENGINE fuzz_manifest.o \
+	-o $OUT/fuzz_manifest  clib-cache.o clib-configure.o clib-settings.o clib-package.o \
+	-I./deps/asprintf -I./deps -I./asprintf \
+	fuzz_lib.a -L/usr/lib/x86_64-linux-gnu -lcurl
+
+echo "[libfuzzer]" > $OUT/fuzz_manifest.options
+echo "detect_leaks=0" >> $OUT/fuzz_manifest.options
+
+# ! mv Pass to out since using relative path
+mv $REPORT_PASS $OUT
```
Comment (EYH0602) [inline projects/clib/build_w_pass.sh]: if compileing with `.o` object files, this `mv` is not needed. Path issue only exists with dynamic libraries (`.so`)

File: projects/eigen/build_w_pass.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# build project
+mkdir build_dir && cd build_dir
+cmake ..
+make install
```
Comment (EYH0602) [inline projects/eigen/build_w_pass.sh:27]: add flags before `make install`

File: projects/assimp/build_w_pass.sh [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+################################################################################
+
+
+# generate build env and build assimp
+cmake CMakeLists.txt -G "Ninja" -DBUILD_SHARED_LIBS=OFF -DASSIMP_BUILD_ZLIB=ON \
+                                -DASSIMP_BUILD_TESTS=OFF -DASSIMP_BUILD_ASSIMP_TOOLS=OFF \
+                                -DASSIMP_BUILD_SAMPLES=OFF
+                                 
+# ! add Pass before building the fazzers
+export REPORT_FLAGS="-Xclang -load -Xclang /src/assimp/ReportFunctionExecutedPass/libreporter.so -flegacy-pass-manager"
```
Comment (EYH0602) [inline projects/assimp/build_w_pass.sh]: in `REPORT_FLAGS`, the `.so` file should be `libReportPass.so`

File: projects/coturn/build_w_pass.sh [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+mkdir my_build
+
+# ! add Pass before building the fazzers
+export REPORT_FLAGS="-Xclang -load -Xclang /src/coturn/ReportFunctionExecutedPass/libReportPass.so -flegacy-pass-manager"
+export CFLAGS="$CFLAGS /src/coturn/ReportFunctionExecutedPass/libreporter.so $REPORT_FLAGS"
```
Comment (EYH0602) [inline projects/coturn/build_w_pass.sh]: missing dependency, should be

```sh
REPORT_FLAGS="-Xclang -load -Xclang $REPORT_PASS/libReportPass.so -flegacy-pass-manager"
REPORTER_FLAGS="$REPORT_PASS/reporter.c++.o -lc++ -pthread -lm"
export CFLAGS="${CFLAGS:=} $REPORT_FLAGS $REPORTER_FLAGS"
export CXXFLAGS="${CXXFLAGS:=} $REPORT_FLAGS $REPORTER_FLAGS"
```

Comment (EYH0602) [reply projects/coturn/build_w_pass.sh]: this might also be the cause of other compile errors

File: projects/augeas/build_w_pass.sh [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+for fuzzer in augeas_api_fuzzer augeas_escape_name_fuzzer augeas_fa_fuzzer; do
+    $CXX $CXXFLAGS -std=c++11 -Isrc/ `xml2-config --cflags` \
+        $fuzzer.cc -o $OUT/$fuzzer $LIB_FUZZING_ENGINE \
+        src/.libs/libaugeas.a src/.libs/libfa.a ./gnulib/lib/.libs/libgnu.a \
+        /usr/lib/x86_64-linux-gnu/libxml2.a
+done
+
+# ! mv Pass to out since using relative path
+mv $REPORT_PASS $OUT
```
Comment (EYH0602) [inline projects/augeas/build_w_pass.sh:47]: can you try `fuzz_w_pass` on this project? 
`CFLAGS` is added using abs-path to `ReportFunctionExecutedPass` and here you change path of `ReportFunctionExecutedPass`. 
removing this line might help if `fuzz_w_pass` does not work.

File: projects/cmark/build_w_pass.sh [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# ! add Pass before building the fazzers
+export REPORT_FLAGS="-Xclang -load -Xclang $REPORT_PASS/libReportPass.so -flegacy-pass-manager -fPIC"
+REPORTER_FLAGS="$REPORT_PASS/libreporter.so -lc++ -pthread -lm -fPIC"
```
Comment (EYH0602) [inline projects/cmark/build_w_pass.sh]: why `-fPIC` is needed here?

File: projects/coturn/build_w_pass.sh [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+cp FuzzStunClient_seed_corpus.zip $OUT/FuzzStunClient_seed_corpus.zip
+popd
+
+pushd /lib/x86_64-linux-gnu/
+mkdir $OUT/lib/
+cp libevent* $OUT/lib/.
+popd
+
+# ! mv Pass to out since using relative path
+mv $REPORT_PASS $OUT
```
Comment (EYH0602) [inline projects/coturn/build_w_pass.sh]: mv is not needed when using static lib `.o`

File: projects/double-conversion/build_w_pass.sh [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+################################################################################
+
+mkdir -p ${WORK}/double-conversion
+cd ${WORK}/double-conversion
+
+cmake -GNinja ${SRC}/double-conversion/
+ninja
+
+# ! add Pass before building the fazzers
+echo "+++++@@@@@@@@@@+++++++++++++@@@@@@@@@@@++++++++"
```
Comment (EYH0602) [inline projects/double-conversion/build_w_pass.sh]: please remove lines used for debugging

File: projects/cmark/build_w_pass.sh (hunk 7 above)
Comment (harrison4ride) [reply projects/cmark/build_w_pass.sh]: This project has not passed the w_pass now. One error message is below:
/usr/bin/ld: /src/cmark/ReportFunctionExecutedPass/reporter.c++.o: relocation R_X86_64_32S against `.bss' can not be used when making a shared object; recompile with -fPIC

Comment (EYH0602) [reply projects/cmark/build_w_pass.sh]: got it

File: projects/augeas/build_w_pass.sh (hunk 6 above)
Comment (harrison4ride) [reply projects/augeas/build_w_pass.sh:47]: Fixed now. it should be working, please check the new version.

**Review by DataCorrupted — APPROVED** [review]

**Pull request merged**